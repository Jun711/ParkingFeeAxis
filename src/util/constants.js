// config
export const LATITUDE_DELTA = 0.005
export const defaultAjaxTimeout = 30000
export const DEFAULT_LATITUDE = 49.2820
export const DEFAULT_LONGITUDE = -123.1171

// location permission constants
export const PERMISSION_REQ_TITLE = 'Parking App Location Permission'
export const PERMISSION_REQ_MSG = 'Parking App needs access to your location to locate nearby parking.'

// toast
export const TOAST_HEADER_Y_OFFSET = 145
export const TOAST_HEADER_X_OFFSET = 0
export const LOCATION_ERROR_MSG_ONE = 'Please enable location service.'
export const LOCATION_ERROR_MSG_TWO = 'Location service is unavailable.'
export const LOCATION_ERROR_DEFAULT_MSG = 'Location request timeout.'

// about
export const APP_NAME = 'Path'
export const APP_VERSION = 'Version 1.0.0'

// time
export const WEEKDAY_9_6 = 'weekdayOffice'
export const WEEKDAY_6_PLUS = 'weekdayAfterOffice'
export const SAT_9_6 = 'satOffice'
export const SAT_6_PLUS = 'satAfterOffice'
export const SUN_9_6 = 'sunOffice'
export const SUN_6_PLUS = 'sunAfterOffice'
export const FREE_PARKING = 'freeParking'
export const FREE_PARKING_LIMIT = 'until 9am'
export const NO_TIME_LIMIT_TEXT = 'no limit'


// min and max rate
export const MIN_PARKING_RATE = 100
export const MAX_PARKING_RATE = -1

// color
export const LOADER_COLOR = '#515c6d'
export const THEME_COLOR = '#38414e'
export const TEXT_COLOR = '#000'
// export const CENTRE_MARKER_COLOR = '#8585ad'
export const CURRENT_MARKER_COLOR = '#98a3b3'
export const CENTRE_MARKER_COLOR ='#515c6d'
export const WHITE_COLOR = '#fff'

// font size
export const TITLE_FONT_SIZE = 20
export const PRIMARY_FONT_SIZE = 18
export const SECONDARY_FONT_SIZE = 14
export const APP_NAME_FONT_SIZE = 15
export const APP_VERSION_FONT_SIZE = 10

// search bar
export const SEARCH_PLACEHOLDER = 'Search'
export const SEARCH_INPUT_KEY = 'headerSearchBar'

// error message
export const CONTACT_INFO_LOADING_ERROR = 'Contact info loading timeout'
export const FAQ_LOADING_ERROR = 'FAQ list loading timeout'
