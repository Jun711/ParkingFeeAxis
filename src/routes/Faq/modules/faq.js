// contains actions and handlers
import update from 'immutability-helper'
import constants from './actionConstants'
import { PermissionsAndroid, Dimensions, ToastAndroid } from 'react-native'
import {} from '../../../util/constants'
import {
  PARKING_FAQS,
} from '../../../util/constants'
import {
  handleLoadFaqList
} from './faqActionHandlers'

//-------------------------------
//Constants
//-------------------------------
const {
  LOAD_FAQ_LIST,
  LOAD_FAQ_LIST_ERROR
} = constants

//-------------------------------
// Utility function
//-------------------------------


//-------------------------------
// Actions
//-------------------------------
export function loadFaqList() {
  return async (dispatch) => {
    try {
      const response = await fetch(PARKING_FAQS)
      const responseJson = await response.json()
      dispatch({
        type: LOAD_FAQ_LIST,
        payload: responseJson
      })
    } catch (error) {
      dispatch({
        type: LOAD_FAQ_LIST_ERROR,
        payload: error
      })
    }
  }
}

//-------------------------------
//Action Handlers
//-------------------------------
const ACTION_HANDLERS = {
  LOAD_FAQ_LIST: handleLoadFaqList
}

//-------------------------------
// Initialization
//-------------------------------
export const initialState = {
  faqLoaded: false,
  faqList: []
}

export function FaqReducer(state = initialState, action) {
  const handler = ACTION_HANDLERS[action.type]

  return handler ? handler(state, action) : state
}