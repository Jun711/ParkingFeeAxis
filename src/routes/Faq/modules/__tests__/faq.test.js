import { loadFaqList } from '../faq';
import constants from '../actionConstants';

const { LOAD_FAQ_LIST } = constants;

it('creates a LOAD_FAQ_LIST action', () => {
  expect(loadFaqList()).toEqual(
    {
      type: LOAD_FAQ_LIST,
    }
  );
  // using Jest snapshot
  expect(loadFaqList()).toMatchSnapshot();
});