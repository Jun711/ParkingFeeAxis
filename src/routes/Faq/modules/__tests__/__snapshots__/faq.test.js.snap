// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FaqList /> should handle LOAD_FAQ_LIST failure action 1`] = `
Array [
  Object {
    "payload": [Error: error],
    "type": "LOAD_FAQ_LIST_ERROR",
  },
]
`;

exports[`<FaqList /> should handle LOAD_FAQ_LIST success action 1`] = `
Array [
  Object {
    "payload": Object {
      "faqs": Array [
        Object {
          "text": "Frequently asked questions",
        },
      ],
    },
    "type": "LOAD_FAQ_LIST",
  },
]
`;
